x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSender.java

```diff
@@ -208,6 +214,27 @@ private void reportFailures(ComputeListener computeListener) {
         }
     }
 
+    private List<ShardSearchFailure> selectFailures() {
+        assert reportedFailure == false;
+        final List<ShardSearchFailure> failures = new ArrayList<>();
+        final Set<Exception> seen = Collections.newSetFromMap(new IdentityHashMap<>());
+        for (Map.Entry<ShardId, ShardFailure> e : shardFailures.entrySet()) {
+            final ShardFailure failure = e.getValue();
+            if (ExceptionsHelper.unwrap(failure.failure(), TaskCancelledException.class) != null) {
```

Comment:
I wonder if it is worth keeping unwrapped exception in `trackShardLevelFailure`?